### PR TITLE
chore(ci): check existing EAS builds before creating a new one

### LINKED ISSUE #14467: Build app from PR
- Add possibility to run the mobile app from PR easily for quick review and testing
- Rebuild native part only if necessary to optimize EAS costs

### PR DESCRIPTION
**Pull request by matejkriz**
## Description

- runtimeVersion should ensure the compatibility, we should not create a new builds with the same runtimeVersion.
- Listing two latest successful builds is fragile, it can list i.e. two android builds if ios had some issue, but should be good enough for now.

## Related Issue

Resolve [#14467](https://github.com/trezor/trezor-suite/issues/14467)

### CODE REVIEW

**Comment by matejkriz** [comment]
I just realized that build-mobile label is not working as expected, because it is still limited to PRs touching whitelisted paths (the path condition also applies for the labeled event) 🤦

File: .github/workflows/build-suite-native-preview.yml [hunk 1]
```diff
@@ -44,7 +46,26 @@ jobs:
...
         run: yarn workspaces focus @suite-native/app
 
+      - name: Check runtimeVersion builds
+        run: |
+          cd suite-native/app
+          # Read runtimeVersion from dynamic app.json
+          RUNTIME_VERSION=$(npx expo config --json | jq -r '.runtimeVersion')
+          echo "Current runtimeVersion is $RUNTIME_VERSION"
+          # Check if there is a build with the same runtimeVersion on EAS already
+          EXISTING_BUILD=$(eas build:list --status=finished  --build-profile=preview --channel=preview --limit=1  --non-interactive --runtime-version=$RUNTIME_VERSION)
```
Comment (Nodonisko) [inline .github/workflows/build-suite-native-preview.yml:56]: Why not list like 20 of them and simply `grep` in it?

**Review by Nodonisko — APPROVED** [review]

File: .github/workflows/build-suite-native-preview.yml (hunk 1 above)
Comment (matejkriz) [reply .github/workflows/build-suite-native-preview.yml:56]: What would be the benefit? There is a filter by --runtime-version, so I believe that 1 existing build is sufficient.

Comment (Nodonisko) [reply .github/workflows/build-suite-native-preview.yml:56]: Oh I didn't noticed that filter then it's fine

**Pull request merged**